### PR TITLE
Features/datepicker

### PR DESCRIPTION
**Pull request by kimhjona**

### CODE REVIEW

**Comment by kimhjona** [comment]
hey @mkulinski ,
there are 72 files (which is a lot), but most of them are an exact clone of the airbnb datepicker. I already removed the testing files since we don't actually use them and can submit a separate PR for them.

**Comment by kimhjona** [comment]
Also, I'll delete the other branch once this one goes through (just in case)

File: example/src/SingleDatePickerWrapper.jsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
+import React from 'react';
```
Comment (mkulinski) [inline example/src/SingleDatePickerWrapper.jsx]: Why is this file here?

File: example/src/DateRangePickerWrapper.jsx [hunk 2]
```diff
@@ -0,0 +1,144 @@
+import React from 'react';
```
Comment (mkulinski) [inline example/src/DateRangePickerWrapper.jsx]: Why is this file here?

File: package.json [hunk 3]
```diff
@@ -132,6 +133,24 @@
     "velocity-animate": "^1.5.0",
     "redux": "3.6.0",
     "redux-immutable-state-invariant": "^2.0.0",
-    "redux-thunk": "2.2.0"
+    "redux-thunk": "2.2.0",
+    "@types/moment": "^2.13.0",
```
Comment (mkulinski) [inline package.json]: Are all of these needed?

File: src/components/DatePicker/components/OutsideClickHandler.jsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      { capture: true },
+    );
+  }
+
+  componentWillUnmount() {
+    if (this.clickHandle) removeEventListener(this.clickHandle);
+  }
+
+  onOutsideClick(e) {
+    // console.log('e:',e);
```
Comment (mkulinski) [inline src/components/DatePicker/components/OutsideClickHandler.jsx]: remove un-needed comments

File: src/components/DatePicker/index.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
+// import undefined from '../../../lib/@types/src/components/DatePicker/utils/toMomentObject';
```
Comment (mkulinski) [inline src/components/DatePicker/index.ts]: removed comments.

File: example/src/DateRangePickerWrapper.jsx (hunk 2 above)
Comment (kimhjona) [reply example/src/DateRangePickerWrapper.jsx]: it's similar to the ReactDataGrid file - you need these two folders as a component to have a working calendar. I guess the reason it's like this is b/c I inherited it like this. 

We can talk about it in an hour or so

File: package.json (hunk 3 above)
Comment (kimhjona) [reply package.json]: def not
